modules/search/client/directives/search.directive.js

```diff
@@ -0,0 +1,280 @@
+			}
+
+			function initializePagination() {
+				// prevent accidental divide by zero
+				self.limit = Math.max(1, self.limit);
+				//set the number of pages so the pagination can update
+				self.numPages = Math.ceil(self.count / self.limit);
+				self.currentPage = Math.ceil(self.start / self.limit) + 1;
+				console.log("currentPage", self.currentPage, self.start);
+				var start = 1;
+				var paginationDisplaySize = 6;
+				var end;
+				var i;
+				start = Math.max(start, self.currentPage - paginationDisplaySize / 2);
+				end = start + paginationDisplaySize;
+				if (end > self.numPages) {
+					end = self.numPages + 1;
+					start = Math.max(1, end - paginationDisplaySize);
+				}
+				self.pages = [];
+				for (i = start; i < end; i++) {
+					self.pages.push(i);
+				}
+			}
+		}
+	};
+}
+
+// -----------------------------------------------------------------------------------
+//
+// DIRECTIVE: Main search widget
+//
+// -----------------------------------------------------------------------------------
+directiveMainSearch.$inject = ['SearchService'];
+/* @ngInject */
+function directiveMainSearch(SearchService) {
+	var directive = {
+		restrict: 'E',
+		scope: {
+			project: '=',
+			preload: '='
+		},
+		controllerAs: 'vm',
+		templateUrl: 'modules/search/client/views/partials/search-widget.html',
+		controller: function ($scope, $rootScope) {
+			var self = this;
+			self.searchText = $scope.preload ? SearchService.getSearchText() : '';
+			self.search = search;
+			self.toggleSearch = toggleSearch;
+			self.searchTextKeyPress = searchTextKeyPress;
+
+			$rootScope.$on('search-text-changed', function () {
+				self.searchText = SearchService.getSearchText();
+			});
+
+			function searchTextKeyPress(event) {
+				if (event.which === 13) {
+					event.preventDefault();
+					if (self.searchText.length > 1) {
+						self.search();
+					}
+				}
+			}
+
+			function search() {
+				SearchService.redirectSearchDocuments($scope.project, self.searchText);
+			}
+
+			function toggleSearch() {
+				$scope.swOpen = !$scope.swOpen;
+			}
+		}
+	};
+	return directive;
+}
+
+
+// -----------------------------------------------------------------------------------
+//
+// DIRECTIVE: Display instructions for user
+//
+// -----------------------------------------------------------------------------------
+directiveModalSearchInstructions.$inject = ['$modal'];
+/* @ngInject */
+function directiveModalSearchInstructions($modal) {
+	var directive = {
+		restrict: 'A',
+		scope: {
+		},
+		link: function (scope, element, attrs) {
+			element.on('click', function () {
+				var modalInstructions = $modal.open({
+					animation: true,
+					templateUrl: 'modules/search/client/views/partials/modal-search-instructions.html',
+					controller: 'controllerModalSearchInstructions',
+					controllerAs: 'instruct',
+					size: 'lg'
+				});
+				modalInstructions.result.then(function (data) {
```

Comment:
Ah you are talking about the .then() ... do nothing ... stuff.  Frankly, this is a case of copy and paste the code from doc manager's instruction button.  I will try to delete the empty .then()